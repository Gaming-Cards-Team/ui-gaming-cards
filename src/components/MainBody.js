import React from "react";
import Banner from "./Banner";
import TopMessage from "./TopMessage";
import CardGallery from "./CardGallery";
import BlankLines from './BlankLines'

const MainBody = () => {
  return (
    <>
      <BlankLines/>
      <Banner/>
      <TopMessage/>
      <CardGallery/>
    </>
  );
};

export default MainBody;
