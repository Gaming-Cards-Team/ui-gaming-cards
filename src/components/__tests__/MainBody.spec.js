import React from 'react';
import MainBody from '../MainBody';

import renderer from 'react-test-renderer';

describe('Test Main Body component', () => {
    it('should validate the component', () => {
        const mainBody = renderer.create(<MainBody />).toJSON();

        expect(mainBody).toMatchSnapshot();
    });
});