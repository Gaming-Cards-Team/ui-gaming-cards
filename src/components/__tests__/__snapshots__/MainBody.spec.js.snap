// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Main Body component should validate the component 1`] = `
Array [
  <br />,
  <br />,
  <br />,
  <img
    alt="Star Wars Banner"
    data-testid="banner"
    src="banner.png"
    width="100%"
  />,
  <div
    style={
      Object {
        "background": "black",
        "color": "white",
        "fontSize": "larger",
        "padding": "10px 30px",
        "width": "100%",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col-9"
        style={
          Object {
            "padding": "10px 0px",
          }
        }
      >
        Are you looking for cards for other game cards? check our catalog 
        <a
          href="{#}"
          style={
            Object {
              "color": "#2d963f",
              "fontWeight": "bold",
            }
          }
        >
          here
        </a>
      </div>
      <div
        className="col-3"
        style={
          Object {
            "textAlign": "right",
          }
        }
      >
        <a
          href="https://www.youtube.com/watch?v=3gPu9pp2WOM"
          rel="noopener noreferrer"
          style={
            Object {
              "color": "#2d963f",
              "fontWeight": "bold",
            }
          }
          target="_blank"
        >
          How to play 
          <img
            alt="Video Logo"
            data-testid="youtubeVideo"
            height="48"
            src="youtubeVideoImage.png"
          />
        </a>
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "paddingBottom": 50,
        "paddingLeft": 100,
        "paddingRight": 100,
        "paddingTop": 30,
      }
    }
  >
    <div>
      <div
        className="card-deck"
      />
      <div
        style={
          Object {
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          aria-busy="true"
          className=""
          style={Object {}}
        >
          <svg
            aria-label="audio-loading"
            fill="green"
            height={80}
            viewBox="0 0 135 140"
            width={80}
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="120"
              rx="6"
              width="15"
              y="10"
            >
              <animate
                attributeName="height"
                begin="0.5s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="120;110;100;90;80;70;60;50;40;140;120"
              />
              <animate
                attributeName="y"
                begin="0.5s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="10;15;20;25;30;35;40;45;50;0;10"
              />
            </rect>
            <rect
              height="120"
              rx="6"
              width="15"
              x="30"
              y="10"
            >
              <animate
                attributeName="height"
                begin="0.25s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="120;110;100;90;80;70;60;50;40;140;120"
              />
              <animate
                attributeName="y"
                begin="0.25s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="10;15;20;25;30;35;40;45;50;0;10"
              />
            </rect>
            <rect
              height="140"
              rx="6"
              width="15"
              x="60"
            >
              <animate
                attributeName="height"
                begin="0s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="120;110;100;90;80;70;60;50;40;140;120"
              />
              <animate
                attributeName="y"
                begin="0s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="10;15;20;25;30;35;40;45;50;0;10"
              />
            </rect>
            <rect
              height="120"
              rx="6"
              width="15"
              x="90"
              y="10"
            >
              <animate
                attributeName="height"
                begin="0.25s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="120;110;100;90;80;70;60;50;40;140;120"
              />
              <animate
                attributeName="y"
                begin="0.25s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="10;15;20;25;30;35;40;45;50;0;10"
              />
            </rect>
            <rect
              height="120"
              rx="6"
              width="15"
              x="120"
              y="10"
            >
              <animate
                attributeName="height"
                begin="0.5s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="120;110;100;90;80;70;60;50;40;140;120"
              />
              <animate
                attributeName="y"
                begin="0.5s"
                calcMode="linear"
                dur="1s"
                repeatCount="indefinite"
                values="10;15;20;25;30;35;40;45;50;0;10"
              />
            </rect>
          </svg>
        </div>
      </div>
    </div>
  </div>,
]
`;
